sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -106,11 +155,13 @@ case class Mode(
     val collationAwareBuffer = child.dataType match {
       case c: StringType if
         !CollationFactory.fetchCollation(c.collationId).supportsBinaryEquality =>
-        val collationId = c.collationId
-        val modeMap = buffer.toSeq.groupMapReduce {
-         case (k, _) => CollationFactory.getCollationKey(k.asInstanceOf[UTF8String], collationId)
+        buffer.toSeq.groupMapReduce {
+         case (k, _) => CollationFactory.getCollationKey(k.asInstanceOf[UTF8String], c.collationId)
         }(x => x)((x, y) => (x._1, x._2 + y._2)).values
-        modeMap
+      case _: ArrayType | _ : StructType if child.dataType.existsRecursively(s =>
```

Comment:
does `Mode` support `MapType`?

Reply:
@uros-db Yes, `Mode` supports all data types. 

Has it been determined whether MapType will be made to support collations? 

<img width="387" alt="image" src="https://github.com/user-attachments/assets/fc8fe896-fa2d-49db-8293-5ceeaa7936d1">

Reply:
@uros-db

Reply:
I see, then let's make sure that we add a comment saying that MapType support is TBD, and also let's catch MapType with collated strings here - in order to throw appropriate error (just like we are doing currently for all complex types)

Reply:
@uros-db It turns out that collation support has been added for maptype. I did not realize --  So I went adhead and added support for collated maptypes to Mode, for this PR. Adding a unit test. Let me know what you think.

Reply:
Yes, `MapType` already has collation support in Spark, and `mode` seems to work on maps (unlike joins for example, which don't work on map types) - so please proceed with handling maps in `mode` and of course adding appropriate tests